gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageObjectAttributes.java

```diff
@@ -101,7 +137,7 @@ public FileTime lastModifiedTime() {
    */
   @Override
   public ImmutableMap<String, String> userMetadata() {
-    return ImmutableMap.copyOf(metadata.getOptions().getUserMetadata());
+    return ImmutableMap.copyOf(info.metadata());
```

Comment:
Seems like `info` could be `null`. Also BlobInfo is immutable and any value that it returns (including collections, maps) is immutable.  Generally we do return empty collection instead of `null` but unfortunately that does not seem to be the case for `BlobInfo.metadata`. /cc @mziccard